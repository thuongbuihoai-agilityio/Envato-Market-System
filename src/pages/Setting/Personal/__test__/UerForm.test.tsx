import { render, screen, waitFor } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

// Components
import UserForm from '@app/pages/Setting/Personal';
import userEvent from '@testing-library/user-event';

const client = new QueryClient();

const setup = () =>
  render(
    <QueryClientProvider client={client}>
      <UserForm />
    </QueryClientProvider>,
  );

describe('Personal Page test cases', () => {
  it('renders UserForm with content', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });

  it('should handle submit form', async () => {
    setup();

    const mockInput = 'mock input';

    const firstNameInput = screen.getByRole<HTMLInputElement>('textbox', {
      name: /first name/i,
      hidden: true,
    });

    const lastNameInput = screen.getByRole<HTMLInputElement>('textbox', {
      name: /last name/i,
      hidden: true,
    });

    const emailInput = screen.getByRole<HTMLInputElement>('textbox', {
      name: /email/i,
      hidden: true,
    });

    const phoneInput = screen.getByRole<HTMLInputElement>('textbox', {
      name: /phone number \(optional\)/i,
      hidden: true,
    });

    const countryInput = screen.getByRole<HTMLInputElement>('textbox', {
      name: /country and region/i,
      hidden: true,
    });

    const cityInput = screen.getByRole<HTMLInputElement>('textbox', {
      name: /city/i,
      hidden: true,
    });

    const addressInput = screen.getByRole<HTMLInputElement>('textbox', {
      name: /address/i,
      hidden: true,
    });

    const postalCodeInput = screen.getByRole<HTMLInputElement>('textbox', {
      name: /postal code/i,
      hidden: true,
    });

    const facebookInput = screen.getByRole<HTMLInputElement>('textbox', {
      name: /facebook/i,
      hidden: true,
    });

    const twitterInput = screen.getByRole<HTMLInputElement>('textbox', {
      name: /twitter/i,
      hidden: true,
    });

    const linkedInInput = screen.getByRole<HTMLInputElement>('textbox', {
      name: /linkedin/i,
      hidden: true,
    });

    const youtubeInput = screen.getByRole<HTMLInputElement>('textbox', {
      name: /youtube/i,
      hidden: true,
    });

    await userEvent.type(firstNameInput, mockInput);
    await userEvent.type(lastNameInput, mockInput);
    await userEvent.type(emailInput, 'mockmail@gmail.com');
    await userEvent.type(phoneInput, '01234567890');
    await userEvent.type(countryInput, mockInput);
    await userEvent.type(cityInput, mockInput);
    await userEvent.type(addressInput, mockInput);
    await userEvent.type(postalCodeInput, '01234567890');
    await userEvent.type(facebookInput, mockInput);
    await userEvent.type(twitterInput, mockInput);
    await userEvent.type(linkedInInput, mockInput);
    await userEvent.type(youtubeInput, mockInput);
    await userEvent.type(firstNameInput, mockInput);

    const submitBtn = screen.getByRole<HTMLButtonElement>('button', {
      name: /btn\-save\-profile/i,
      hidden: true,
    });

    await userEvent.click(submitBtn);

    waitFor(() => {
      expect(submitBtn.disabled).toBeTruthy();
    });
  });
});
