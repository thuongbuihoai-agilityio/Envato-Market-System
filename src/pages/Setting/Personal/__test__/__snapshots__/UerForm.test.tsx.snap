// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Personal Page test cases renders UserForm with content 1`] = `
<div>
  <form
    class="chakra-stack css-tl3ftk"
    id="register-form"
  >
    <div
      class="css-x1jqhu"
    >
      <section
        class="css-17s3n79"
      >
        <h3
          class="chakra-heading css-r8aai1"
        >
          personal information's
        </h3>
        <div
          class="chakra-stack css-1s7rg7j"
        >
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <label
              class="chakra-form__label css-yeb7cv"
              for="field-:r0:"
              id="field-:r0:-label"
            >
              First Name
            </label>
            <div
              class="chakra-input__group css-169ufh8"
              data-group="true"
            >
              <input
                class="chakra-input css-r4nbqm"
                id="field-:r0:"
                name="firstName"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <label
              class="chakra-form__label css-yeb7cv"
              for="field-:r1:"
              id="field-:r1:-label"
            >
              Last Name
            </label>
            <div
              class="chakra-input__group css-169ufh8"
              data-group="true"
            >
              <input
                class="chakra-input css-r4nbqm"
                id="field-:r1:"
                name="lastName"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="chakra-stack css-rluqv0"
        >
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <label
              class="chakra-form__label css-yeb7cv"
              for="field-:r2:"
              id="field-:r2:-label"
            >
              Email
            </label>
            <div
              class="chakra-input__group css-169ufh8"
              data-group="true"
            >
              <input
                class="chakra-input css-r4nbqm"
                id="field-:r2:"
                name="email"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <label
              class="chakra-form__label css-yeb7cv"
              for="field-:r3:"
              id="field-:r3:-label"
            >
              Phone Number (optional)
            </label>
            <div
              class="chakra-input__group css-169ufh8"
              data-group="true"
            >
              <input
                class="chakra-input css-r4nbqm"
                id="field-:r3:"
                name="phoneNumber"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <h2
          class="chakra-heading css-7yzvk1"
        >
          Personal Address
        </h2>
        <div
          class="chakra-stack css-1fq11yp"
        >
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <label
              class="chakra-form__label css-yeb7cv"
              for="field-:r4:"
              id="field-:r4:-label"
            >
              Country and Region
            </label>
            <div
              class="chakra-input__group css-169ufh8"
              data-group="true"
            >
              <input
                class="chakra-input css-r4nbqm"
                id="field-:r4:"
                name="country"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <label
              class="chakra-form__label css-yeb7cv"
              for="field-:r5:"
              id="field-:r5:-label"
            >
              City
            </label>
            <div
              class="chakra-input__group css-169ufh8"
              data-group="true"
            >
              <input
                class="chakra-input css-r4nbqm"
                id="field-:r5:"
                name="city"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="chakra-stack css-fskqe1"
        >
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <label
              class="chakra-form__label css-yeb7cv"
              for="field-:r6:"
              id="field-:r6:-label"
            >
              Address
            </label>
            <div
              class="chakra-input__group css-169ufh8"
              data-group="true"
            >
              <input
                class="chakra-input css-r4nbqm"
                id="field-:r6:"
                name="address"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <label
              class="chakra-form__label css-yeb7cv"
              for="field-:r7:"
              id="field-:r7:-label"
            >
              Postal Code
            </label>
            <div
              class="chakra-input__group css-169ufh8"
              data-group="true"
            >
              <input
                class="chakra-input css-r4nbqm"
                id="field-:r7:"
                name="postalCode"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <h2
          class="chakra-heading css-7yzvk1"
        >
          Social Information
        </h2>
        <div
          class="chakra-stack css-1fq11yp"
        >
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <label
              class="chakra-form__label css-yeb7cv"
              for="field-:r8:"
              id="field-:r8:-label"
            >
              Facebook
            </label>
            <div
              class="chakra-input__group css-169ufh8"
              data-group="true"
            >
              <input
                class="chakra-input css-r4nbqm"
                id="field-:r8:"
                name="facebookURL"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <label
              class="chakra-form__label css-yeb7cv"
              for="field-:r9:"
              id="field-:r9:-label"
            >
              TWitter
            </label>
            <div
              class="chakra-input__group css-169ufh8"
              data-group="true"
            >
              <input
                class="chakra-input css-r4nbqm"
                id="field-:r9:"
                name="twitterURL"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="chakra-stack css-fskqe1"
        >
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <label
              class="chakra-form__label css-yeb7cv"
              for="field-:ra:"
              id="field-:ra:-label"
            >
              Linkedin
            </label>
            <div
              class="chakra-input__group css-169ufh8"
              data-group="true"
            >
              <input
                class="chakra-input css-r4nbqm"
                id="field-:ra:"
                name="linkedinURL"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <label
              class="chakra-form__label css-yeb7cv"
              for="field-:rb:"
              id="field-:rb:-label"
            >
              Youtube
            </label>
            <div
              class="chakra-input__group css-169ufh8"
              data-group="true"
            >
              <input
                class="chakra-input css-r4nbqm"
                id="field-:rb:"
                name="youtubeURL"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="css-t110l1"
        >
          <p
            class="chakra-text css-dz8804"
          />
          <div
            class="css-tz0gqg"
          >
            <button
              aria-label="btn-save-profile"
              class="chakra-button css-glk3as"
              disabled=""
              form="register-form"
              type="submit"
            >
              Save Profile
            </button>
          </div>
        </div>
      </section>
      <div
        class="css-hf8gyw"
      >
        <div
          class="css-0"
        >
          <h4
            class="chakra-heading css-1th4bpb"
          >
            update profile
          </h4>
          <p
            class="chakra-text css-12zxwfm"
          >
            Profile at least Size
            <span
              class="chakra-text css-1999y4e"
            >
              300x300.
            </span>
            Gift to work too.
          </p>
          <div
            class="css-4bwcy0"
          >
            <div
              class="chakra-skeleton css-183zf81"
            >
              <img
                alt="This is the big avatar"
                class="chakra-image css-opaydd"
                src=""
              />
            </div>
            <div
              class="chakra-input__group css-eh9zni"
              data-group="true"
            >
              <div
                class="chakra-input__left-element css-o84xzr"
              >
                <label
                  class="chakra-form__label css-uvho5g"
                  for="file"
                >
                  <img
                    alt="Edit icon"
                    class="chakra-image css-1eh5dcz"
                    src="/icons/edit.svg"
                  />
                </label>
                <input
                  class="chakra-input css-kmhx47"
                  data-testid="upload-image"
                  id="file"
                  name="file"
                  type="file"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
